hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -294,6 +294,7 @@ struct Vgen {
   void emit(const andli& i) { a->And(W(i.d), W(i.s1), i.s0.l(), UF(i.fl)); }
   void emit(const andq& i) { a->And(X(i.d), X(i.s1), X(i.s0), UF(i.fl)); }
   void emit(const andqi& i) { a->And(X(i.d), X(i.s1), i.s0.q(), UF(i.fl)); }
+  void emit(const andzi& i) { a->And(X(i.d), X(i.s1), i.s0.Q(), UF(i.fl)); }
```

Comment:
In https://github.com/facebook/hhvm/pull/7851, @cmuellner just used `q()`, so I think that should be fine here, and you can revert the implementation of `Q()`.